lib/semantics/check-omp-structure.cc

```diff
@@ -98,6 +98,72 @@ void OmpStructureChecker::Leave(const parser::OpenMPBlockConstruct &) {
   ompContext_.pop_back();
 }
 
+void OmpStructureChecker::Enter(const parser::OpenMPSectionsConstruct &x) {
+  const auto &dir{std::get<parser::Verbatim>(x.t)};
+  OmpContext ct{dir.source};
+  ompContext_.push_back(ct);
+  SetContextDirectiveEnum(OmpDirective::SECTIONS);
+  OmpClauseSet allowed{OmpClause::PRIVATE, OmpClause::FIRSTPRIVATE,
+      OmpClause::LASTPRIVATE, OmpClause::REDUCTION};
+  SetContextAllowed(allowed);
+}
+
+void OmpStructureChecker::Leave(const parser::OpenMPSectionsConstruct &x) {
+  ompContext_.pop_back();
+}
+
+void OmpStructureChecker::Enter(const parser::OmpSection &x) {
+  const auto &dir{x.v};  // Verbatim
+  OmpContext ct{dir.source};
+  ompContext_.push_back(ct);
+  SetContextDirectiveEnum(OmpDirective::SECTION);
+
+  if (ompContext_.size() == 1) {
+    context_.Say(
+        dir.source, "Orphaned SECTION directive is prohibited"_err_en_US);
```

Comment:
What does "orphaned" mean? Why isn't this just the same message as below, i.e. "SECTION directive must appear within the SECTIONS construct"?

Reply:
```
 !$omp section  <- Orphaned
  a = 0.0
  !$omp sections
  !$omp section
  a = 3.14
  !$omp end sections
```
I think you're right, the error msg applies to this case too. I will remove that.

Reply:
`GetPrevContext` is removed based on Jean's comment.